src/rules/prefer-invoke.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @fileoverview Rule to check if a call to map should be a call to invokeMap
+ */
+"use strict"
+const _ = require("lodash")
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {}
+  },
+
+  create(context) {
+    const { getLodashMethodVisitors } = require("../util/lodashUtil")
+
+
+    const getParentIfStatement = n => {
+      if (_.get(n, 'parent.type') === "IfStatement") {
+        return n.parent
+      } else if (_.get(n, "parent.parent.type") === "IfStatement") {
+        return n.parent.parent
+      }
+    }
+
+    const getIsFunctionArgument = node => 
+      _.get(node, 'arguments[0].name') || _.get(node, 'arguments[0].property.name')
```

Comment:
This would make this code return a false positive:
```js
if(_.isFunction(x.a)) {
  y.a()
}
```

Reply:
Added it as a valid case.